src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -806,4 +783,76 @@ final case class AgreementApiServiceImpl(
         .whenA(activeAgreement.nonEmpty)
     } yield ()
   }
+
+  override def addAgreementConsumerDocument(agreementId: String, documentSeed: DocumentSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerDocument: ToEntityMarshaller[Document],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE) {
+
+    logger.info(s"Adding a consumer document to agreement $agreementId")
+
+    val result: Future[Document] = for {
+      agreementUUID <- agreementId.toFutureUUID
+      document      <- agreementManagementService.addConsumerDocument(
+        agreementUUID,
+        AgreementManagement.DocumentSeed(
+          name = documentSeed.name,
+          prettyName = documentSeed.prettyName,
+          contentType = documentSeed.contentType,
+          path = documentSeed.path
+        )
+      )
+    } yield document.toApi
+
+    onComplete(result) {
+      handleAddDocumentError(s"Error adding a consumer document to agreement $agreementId") orElse {
+        case Success(document) =>
+          addAgreementConsumerDocument200(document)
+      }
+    }
+  }
+
+  override def getAgreementConsumerDocument(agreementId: String, documentId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerDocument: ToEntityMarshaller[Document],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE) {
+
+    logger.info(s"Getting consumer document $documentId from agreement $agreementId")
+
+    val result: Future[Document] = for {
+      agreementUUID <- agreementId.toFutureUUID
+      documentUUID  <- documentId.toFutureUUID
+      document      <- agreementManagementService.getConsumerDocument(agreementUUID, documentUUID)
+    } yield document.toApi
+
+    onComplete(result) {
+      handleGetDocumentError(s"Error getting consumer document $documentId from agreement $agreementId") orElse {
+        case Success(document) =>
+          getAgreementConsumerDocument200(document)
+      }
+    }
+  }
+
+  override def removeAgreementConsumerDocument(agreementId: String, documentId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE) {
+
+    logger.info(s"Removing consumer document $documentId from agreement $agreementId")
+
+    val result: Future[Unit] = for {
+      agreementUUID <- agreementId.toFutureUUID
+      documentUUID  <- documentId.toFutureUUID
+      result        <- agreementManagementService.removeConsumerDocument(agreementUUID, documentUUID)
```

Comment:
🚓 we need to check that the requester is actually the agreement consumer (and maybe that the agreement is in a specific state? I guess it should not be possible to remove documents from an active agreement?)